text/0431-sagemaker-l2-endpoint.md

```diff
@@ -0,0 +1,1150 @@
+SageMaker Model Hosting L2 Constructs
+
+* **Original Author(s):** @pvanlund, @mattmcclean, @l2yao, @jetterdj, @foxpro24, @rangoju
+* **Tracking Issue**: #431
+* **API Bar Raiser**: *TBD*
+
+This feature supports the creation of Amazon SageMaker real-time inference hosted endpoints using a
+new set of L2 constructs for the `Endpoint`, `EndpointConfig`, and `Model` CloudFormation resources.
+
+## Working Backwards
+
+### CHANGELOG
+
+`feat(sagemaker): add model hosting L2 constructs`
+
+### README
+
+---
+
+# Amazon SageMaker Construct Library
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![cfn-resources: Stable](https://img.shields.io/badge/cfn--resources-stable-success.svg?style=for-the-badge)
+
+> All classes with the `Cfn` prefix in this module ([CFN Resources]) are always stable and safe to use.
+>
+> [CFN Resources]: https://docs.aws.amazon.com/cdk/latest/guide/constructs.html#constructs_lib
+
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+> The APIs of higher level constructs in this module are experimental and under active development.
+> They are subject to non-backward compatible changes or removal in any future version. These are
+> not subject to the [Semantic Versioning](https://semver.org/) model and breaking changes will be
+> announced in the release notes. This means that while you may use them, you may need to update
+> your source code when upgrading to a newer version of this package.
+
+---
+
+<!--END STABILITY BANNER-->
+
+Amazon SageMaker provides every developer and data scientist with the ability to build, train, and
+deploy machine learning models quickly. Amazon SageMaker is a fully-managed service that covers the
+entire machine learning workflow to label and prepare your data, choose an algorithm, train the
+model, tune and optimize it for deployment, make predictions, and take action. Your models get to
+production faster with much less effort and lower cost.
+
+## Installation
+
+Install the module:
+
+```console
+$ npm i @aws-cdk/aws-sagemaker
+```
+
+Import it into your code:
+
+```typescript
+import * as sagemaker from '@aws-cdk/aws-sagemaker';
+```
+
+## Model
+
+By creating a model, you tell Amazon SageMaker where it can find the model components. This includes
```

Comment:
I don't know yet what a model is, and did you meant to say "by creating a `Model`" ? As in, refer to a class here?

Reply:
Most of the current verbiage in the README was taken from SageMaker's public documentation ([link relevant to this section](https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-deployment.html#realtime-endpoints-deployment-create-model)), which uses "*model*" to describe both the SageMaker resource and its associated abstract ML concept.

Which of the following are you looking for?

1. Use/paraphrase a one-liner from the [AWS glossary](https://docs.aws.amazon.com/general/latest/gr/glos-chap.html#ml-model), as in the following
    > In machine learning (ML), a mathematical model that generates predictions by finding patterns in data.
1. A more substantial write-up where I describe ML concepts to non-ML customers
1. Simply use `Model` in place of the first use of "*model*"?

Reply:
FYI: As I had already expanded the wording a bit more (to provide a bit more ML related context), the version in the next revision will be slightly longer.